test/integration/query-interface/changeColumn.test.js

```diff
@@ -134,115 +134,116 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
-          expect(firstForeignKeys[0].columnName).to.be.equal(newForeignKeys[0].columnName);
+        await this.queryInterface.changeColumn('users', 'level_id', {
+          type: DataTypes.INTEGER,
+          allowNull: true,
+        });
 
-          const describedTable = await this.queryInterface.describeTable({
-            tableName: 'users',
-          });
+        const newForeignKeys = await this.queryInterface.getForeignKeyReferencesForTable('users');
+        expect(firstForeignKeys.length).to.be.equal(newForeignKeys.length);
+        expect(firstForeignKeys[0].columnName).to.be.equal('level_id');
+        expect(firstForeignKeys[0].columnName).to.be.equal(newForeignKeys[0].columnName);
 
-          expect(describedTable.level_id).to.have.property('allowNull');
-          expect(describedTable.level_id.allowNull).to.not.equal(firstTable.level_id.allowNull);
-          expect(describedTable.level_id.allowNull).to.be.equal(true);
+        const describedTable = await this.queryInterface.describeTable({
+          tableName: 'users',
         });
-        if (!['db2', 'ibmi'].includes(dialect)) {
-          it('should change the comment of column', async function () {
-            const describedTable = await this.queryInterface.describeTable({
-              tableName: 'users',
-            });
 
-            expect(describedTable.level_id.comment).to.be.equal(null);
+        expect(describedTable.level_id).to.have.property('allowNull');
+        expect(describedTable.level_id.allowNull).to.not.equal(firstTable.level_id.allowNull);
+        expect(describedTable.level_id.allowNull).to.be.equal(true);
+      });
 
-            await this.queryInterface.changeColumn('users', 'level_id', {
-              type: DataTypes.INTEGER,
-              comment: 'FooBar',
-            });
+      if (!['db2', 'ibmi', 'sqlite'].includes(dialect)) {
+        it('should change the comment of column', async function () {
+          const describedTable = await this.queryInterface.describeTable({
+            tableName: 'users',
+          });
 
-            const describedTable2 = await this.queryInterface.describeTable({ tableName: 'users' });
-            expect(describedTable2.level_id.comment).to.be.equal('FooBar');
+          expect(describedTable.level_id.comment).to.be.equal(null);
+
+          await this.queryInterface.changeColumn('users', 'level_id', {
+            type: DataTypes.INTEGER,
+            comment: 'FooBar',
           });
-        }
-      });
-    }
 
+          const describedTable2 = await this.queryInterface.describeTable({ tableName: 'users' });
+          expect(describedTable2.level_id.comment).to.be.equal('FooBar');
+        });
+      }
+    });
+
+    // sqlite has limited ALTER TABLE capapibilites which requires a workaround involving recreating tables.
+    // This leads to issues with losing data or losing foreign key references.
+    // The tests below address these problems
```

Comment:
I think we can run these tests on all dialects (and only disable the dialects that fail)

Reply:
I have problems with porting these test cases for other dialects. I have described the problem in another comment below (as a reply to another suggestion of yours). I am not well versed within the sequelize code base so I would love if you can provide a way to port the tests for other dialects.

Reply:
I'm also currently working on `changeColumn` to fix many more bugs, I'll take a look at making these tests run in all dialects in my PR because I first need to fix a bunch more bugs in `showConstraint`, `describeTable` etc

Can you add a `// TODO: run in all dialects` above this condition?